commit fed286110f4bab01f93f06c32951fbc120fb71b1
Author: Daniel Borkmann <daniel@iogearbox.net>
Date:   Wed Dec 11 11:28:59 2013 +0100

    crypto: arch - use crypto_memneq instead of memcmp
    
    Replace remaining occurences (just as we did in crypto/) under arch/*/crypto/
    that make use of memcmp() for comparing keys or authentication tags for
    usage with crypto_memneq(). It can simply be used as a drop-in replacement
    for the normal memcmp().
    
    Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
    Cc: James Yonan <james@openvpn.net>
    Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>

diff --git a/arch/s390/crypto/des_s390.c b/arch/s390/crypto/des_s390.c
index bcca01c9989d..200f2a1b599d 100644
--- a/arch/s390/crypto/des_s390.c
+++ b/arch/s390/crypto/des_s390.c
@@ -237,9 +237,9 @@ static int des3_setkey(struct crypto_tfm *tfm, const u8 *key,
 	struct s390_des_ctx *ctx = crypto_tfm_ctx(tfm);
 	u32 *flags = &tfm->crt_flags;
 
-	if (!(memcmp(key, &key[DES_KEY_SIZE], DES_KEY_SIZE) &&
-	    memcmp(&key[DES_KEY_SIZE], &key[DES_KEY_SIZE * 2],
-		   DES_KEY_SIZE)) &&
+	if (!(crypto_memneq(key, &key[DES_KEY_SIZE], DES_KEY_SIZE) &&
+	    crypto_memneq(&key[DES_KEY_SIZE], &key[DES_KEY_SIZE * 2],
+			  DES_KEY_SIZE)) &&
 	    (*flags & CRYPTO_TFM_REQ_WEAK_KEY)) {
 		*flags |= CRYPTO_TFM_RES_WEAK_KEY;
 		return -EINVAL;